packages/commonwealth/client/scripts/hooks/useSearchResults.ts

```diff
@@ -0,0 +1,66 @@
+import { useMemo } from 'react';
+import { useDebounce } from 'usehooks-ts';
+
+import { APIOrderBy, APIOrderDirection } from '../helpers/constants';
+import { useSearchThreadsQuery } from '../state/api/threads';
+import { useSearchCommentsQuery } from '../state/api/comments';
+import { useSearchChainsQuery } from '../state/api/chains';
+import { useSearchProfilesQuery } from '../state/api/profiles';
+import { SearchScope } from '../models/SearchQuery';
+import app from '../state';
+
+const NUM_RESULTS_PER_SECTION = 2;
+
+type Filter = 'threads' | 'replies' | 'communities' | 'members';
+
+const useSearchResults = (
+  searchTerm: string,
+  filters: Filter[],
+  resultsPerSection?: number
+) => {
+  const chainId = app.activeChainId() || 'all_chains';
+  const debouncedSearchTerm = useDebounce<string>(searchTerm, 500);
+
+  const sharedQueryOptions = {
+    chainId,
+    searchTerm: debouncedSearchTerm,
+    limit: resultsPerSection ? resultsPerSection : NUM_RESULTS_PER_SECTION,
+    orderBy: APIOrderBy.Rank,
+    orderDirection: APIOrderDirection.Desc,
+  };
+  const queryEnabled = debouncedSearchTerm.length > 0;
+
+  const { data: threadsData } = useSearchThreadsQuery({
+    ...sharedQueryOptions,
+    enabled: queryEnabled,
+  });
+
+  const { data: commentsData } = useSearchCommentsQuery({
+    ...sharedQueryOptions,
+    enabled: queryEnabled,
+  });
+
+  const { data: chainsData } = useSearchChainsQuery({
+    ...sharedQueryOptions,
+    enabled: queryEnabled,
+  });
+
+  const { data: profilesData } = useSearchProfilesQuery({
+    ...sharedQueryOptions,
+    includeRoles: false,
+    enabled: queryEnabled,
+  });
+
+  const searchResults = useMemo(() => {
+    return {
+      [SearchScope.Threads]: filters.includes("threads") ? threadsData?.pages?.[0]?.results || [] : [],
+      [SearchScope.Replies]: filters.includes("replies") ? commentsData?.pages?.[0]?.results || [] : [],
+      [SearchScope.Communities]: filters.includes("communities") ? chainsData?.pages?.[0]?.results || [] : [],
+      [SearchScope.Members]: filters.includes("members") ? profilesData?.pages?.[0]?.results || [] : [],
```

Comment:
if you do `pages[0]`, does it mean we always display only first page?

Reply:
I did a test passing a large number to [useSearchResults hook](https://github.com/hicommonwealth/commonwealth/blob/daniel.4119.search-bar/packages/commonwealth/client/scripts/hooks/useSearchResults.ts), searching for the term `'a'` and these are the results brought:

![image](https://github.com/hicommonwealth/commonwealth/assets/30223098/f58d8666-a54c-4b0c-b6d4-f00459738654)

![image](https://github.com/hicommonwealth/commonwealth/assets/30223098/33a65ba4-f6b6-4f74-9345-f14f6d0dc8bb)

So the results for every section (i.e. 'threads', 'replies', 'communities' or 'members') are inside `pages[0]`.
It seems that this data structure was designed like this. Here is another example of the results being read from `data.pages[0].results` array: https://github.com/hicommonwealth/commonwealth/blob/master/packages/commonwealth/client/scripts/views/pages/manage_community/ManageCommunityPage.tsx#L64